pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -106,43 +106,19 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 	r.reqLogger = log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	r.reqLogger.Info("Reconciling ClusterDeployment")
 
-	// Fetch the ClusterDeployment instance
-	instance := &hivev1.ClusterDeployment{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
-	}
+	processCD, instance, err := utils.CheckClusterDeployment(request, r.client, r.reqLogger)
 
 	if instance.DeletionTimestamp != nil {
 		return r.handleDelete(request, instance)
 	}
 
-	// Just return if this is not a managed cluster
-	if val, ok := instance.Labels[config.ClusterDeploymentManagedLabel]; ok {
-		if val != "true" {
-			r.reqLogger.Info("Is not a managed cluster")
-			return r.handleDelete(request, instance)
-		}
-	} else {
-		// Managed tag is not present which implies it is not a managed cluster
-		r.reqLogger.Info("Is not a managed cluster")
-		return r.handleDelete(request, instance)
-	}
-
-	// Return if alerts are disabled on the cluster
-	if _, ok := instance.Labels[config.ClusterDeploymentNoalertsLabel]; ok {
-		r.reqLogger.Info("Managed cluster with Alerts disabled", "Namespace", request.Namespace, "Name", request.Name)
+	// If we don't manage this cluster: log, delete, return
+	if !processCD {
+		r.reqLogger.Info("ClusterDeployment isn't watched by this operator:  " + config.ClusterDeploymentManagedLabel + "=" + instance.GetLabels()[config.ClusterDeploymentManagedLabel] + ", " + config.ClusterDeploymentNoalertsLabel + "=" + instance.GetLabels()[config.ClusterDeploymentNoalertsLabel])
```

Comment:
Do we want to log everytime we see a clusterDeployment that we don't manage? Also I think this line would read easier if we created a new var `proccessCDMsg` (or something) and using `fmt.Sprintf()`